site/public/js/server.js

```diff
@@ -182,7 +182,20 @@ function newDeleteCourseMaterialForm(id, file_name) {
 
     $('.popup-form').css('display', 'none');
     var form = $("#delete-course-material-form");
-    $('.delete-course-material-message', form).text(file_name);
+    $('.delete-course-material-message', form).html('');
+    let cm_message = '<b>'+file_name+'</b>';
+    if (str_id !== null) {
+        const files_or_links = $(`[id^=file_viewer_${str_id}]`);
+        const num_of_links = files_or_links.filter(function () {
+            return (($(this).siblings('.file-viewer').children('a[data-is-link="1"]').length) === 1);
+        }).length;
+        const num_of_files = files_or_links.length - num_of_links;
+        const file_s = (num_of_files > 1) ? 's' : '';
+        const link_s = (num_of_links > 1) ? 's' : '';
+        const num_links_txt = (num_of_links === 0) ? '</em>)' : ` and <b>${num_of_links}</b> link${link_s}</em>)`
+        cm_message += ` (<em>contains <b>${num_of_files}</b> file${file_s}` + num_links_txt;
+    }
+    $('.delete-course-material-message', form).html(cm_message);
```

Comment:
Doing this introduces the possibility for XSS attacks.  The likelihood of one instructor attacking another via the filename is extraordinarily low of course, but it's generally bad practice to use `.html()`.  It would be better to use `document.createElement()` or similar instead.

Reply:
jQuery also provides a neat way to create an element along with allowing to specify inner text, see [this commit](https://github.com/Submitty/Submitty/pull/8615/commits/94e8b7beb2b772adb98864647ca5adc2aaf67118) from one of my PRs for example. Using [setHTML()](https://developer.mozilla.org/en-US/docs/Web/API/Element/setHTML) is probably the simplest but isn't recommended as it is still experimental and support across browsers isn't satisfactory.